io/src/main/scala/laika/io/model/Input.scala

```diff
@@ -71,17 +72,21 @@ case class TextInput[F[_]: Functor] (path: Path, docType: TextDocumentType, inpu
 
 object TextInput {
 
-  private def readAll[F[_]: Sync](reader: Resource[F, Reader], sizeHint: Int): F[String] = 
-   reader.use(InputRuntime.readAll(_, sizeHint))
+  private def readAll[F[_]: Sync](input: fs2.Stream[F, Byte], codec: Codec): F[String] = 
+    input.through(fs2.text.decodeWithCharset(codec.charSet)).compile.string
```

Comment:
```suggestion
  private def readAll[F[_]: Concurrent](input: fs2.Stream[F, Byte], codec: Codec): F[String] = 
    input.through(fs2.text.decodeWithCharset(codec.charSet)).compile.string
```

FS2 has both a `Concurrent` compiler and a `Sync` compiler, but using the `Sync` compiler for a potentially `Concurrent` stream is not ideal (http4s has an internal lint to prevent this "downgrade"). See discussion in https://github.com/typelevel/fs2/pull/2657.

Reply:
Oh, that is quite subtle and somewhat counterintuitive - something I wasn't aware of. I also could not find references to this in the fs2 docs.

Reply:
Yeah you are absolutely right 😕 I guess it's sort of documented here:
https://github.com/typelevel/fs2/blob/9db8a220a07fc5b203eff4d311f53f769fb66a5d/core/shared/src/main/scala/fs2/Compiler.scala#L54

Reply:
(But please open an issue about how we might improve that.)